### DISCUSSION TITLE
How to mask content using primitives?

### DISCUSSION THREAD

**Discussion by ptsouchlos** [typst/typst#575] [Q&A] [answered]
Is there any way to use primitives to mask or crop their body content? 

For example:

```
#circle(radius: 25pt, inset: 0pt, outset: 0pt, fill: none, stroke: none)[
  #set align(center + horizon)
  #rect(fill: blue, width: 50pt, height: 30pt)
]
```

This renders:
![image](https://user-images.githubusercontent.com/6591180/229842427-039b12dc-2b16-42d1-966e-ef614e045116.png)

As you can see, the rectangle is not clipped by it's parent (the `circle` in this case).

**Comment by laurmaedje** [comment]
Not yet

**Reply by ptsouchlos** [reply]
I haven't read all the latest change logs, but is this still not possible?

**Reply by laurmaedje** [reply]
Still not possible.

**Comment by ptsouchlos** [comment, marked as answer]
See https://github.com/typst/typst/discussions/2641
